src/rest-api/class-base-api-controller.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Base API Controller
+ *
+ * @package Parsely
+ * @since   3.17.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\REST_API;
+
+use Parsely\Parsely;
+
+/**
+ * Base API Controller.
+ *
+ * Used to define the namespace, version, and endpoints for an API controller. API controllers
+ * should extend this class and implement the `init` method to register endpoints.
+ *
+ * @since 3.17.0
+ */
+abstract class Base_API_Controller {
+
+	/**
+	 * The API namespace.
+	 *
+	 * @var string|false
+	 */
+	public const NAMESPACE = false;
```

Comment:
Can you write a bit about why `false` was chosen here instead of an empty string?

Reply:
No strong reason, I was experimenting with it, and eventually decided against it, but I didn't updated the `NAMESPACE` const. I will change it to be only a string as it simplifies a bit :)